web-api/src/storage/elastic.rs

```diff
@@ -535,65 +596,67 @@ impl KnnSearchParams<'_> {
     }
 }
 
-fn normalize_scores<K>(mut scores: HashMap<K, f32>) -> HashMap<K, f32>
-where
-    K: Eq + Hash,
-{
+fn normalize_scores<Id>(mut scores: ScoreMap<Id>) -> ScoreMap<Id> {
     let max_score = scores
         .values()
+        .map(|(score, _)| score)
         .max_by(|l, r| l.total_cmp(r))
         .copied()
         .unwrap_or_default();
 
-    for score in scores.values_mut() {
+    for (score, _) in scores.values_mut() {
         *score /= max_score;
     }
 
     scores
 }
 
-fn normalize_scores_if_max_gt_1<K>(mut scores: HashMap<K, f32>) -> HashMap<K, f32>
+fn normalize_scores_if_max_gt_1<K, M>(mut scores: HashMap<K, (f32, M)>) -> HashMap<K, (f32, M)>
```

Comment:
:art: we could reuse the `ScoreMap` type alias, it is a bit more restrictive that what is currently there, but we actually don't need the flexibility of the added generics.

:art: we can also rename the `K` to `Id` as you did in another function below
```suggestion
fn normalize_scores_if_max_gt_1<Id>(mut scores: ScoreMap<Id>) -> ScoreMap<Id>
```